syntax = "proto3";

package phenotypes;

message Phenotype {
  uint32 size = 1; // How many cubes
  uint32 dim1 = 2; // Widest dimension
  uint32 dim2 = 3; // 2nd widest dimension
  uint32 dim3 = 4; // 3rd widest dimension
  repeated string rules = 5; // Make this hex int?
  uint32 id = 6;
}

message Dataset {
  map<string, Phenotype> phenotypes = 1;
  uint32 nColors = 2;
  uint32 nCubeTypes = 3;
  uint32 nDimensions = 4;
  uint32 nRules = 5;
  uint32 nTries = 6;
  string assemblyMode = 7;
  string inputPath = 8;
  string pid = 9;
}